Bizware/BizHawk.Bizware.BizwareGL.SlimDX/IGL_SlimDX9.cs

```diff
@@ -518,19 +515,15 @@ public Pipeline CreatePipeline(VertexLayout vertexLayout, Shader vertexShader, S
 
 		public void FreePipeline(Pipeline pipeline)
 		{
-			var pw = pipeline.Opaque as PipelineWrapper;
-
-			pw.VertexDeclaration.Dispose();
-			pw.FragmentShader.IGLShader.Release();
-			pw.VertexShader.IGLShader.Release();
+			using (pipeline.Opaque as PipelineWrapper) { }

```

Comment:
You didn't even fix the bug you claim to be fixing, you just removed the failing calls entirely.

Reply:
> you just removed the failing calls entirely.

I'm not! Code was moved to PipelineWrapper class look:
`		class PipelineWrapper : IDisposable // Disposable fields cleaned up by FreePipeline
		{
			public d3d9.VertexDeclaration VertexDeclaration;
			public ShaderWrapper VertexShader, FragmentShader;
			public int VertexStride;
			public d3d9.ConstantTable fsct, vsct;

			public void Dispose()
			{
				using (VertexDeclaration)
				{
					using (fsct)
						FragmentShader.IGLShader.Release();
					using (vsct)
						VertexShader.IGLShader.Release();
				}
			}
		}`

Btw how using works (I know that student could be confused reading code like this, but he/she must know how it works!). Just look:
https://ufile.io/1unmuefh

Reply:
I'll give you fixing it, but I still think it should be `?.`.